packages/conf-rocksdb-ge-5-17-2/conf-rocksdb-ge-5-17-2.5.17.2/opam

```diff
@@ -0,0 +1,37 @@
+opam-version: "2.0"
+homepage: "http://rocksdb.org/"
+maintainer: "Chet Murthy <chetsky@gmail.com>"
+x-maintenance-intent: [ "(latest)" ]
+authors: "Chet Murthy (copied from Anton Bachin)"
+bug-reports: "https://github.com/chetmurthy/ocaml-rocksdb/issues"
+license: "Apache-2.0"
+
+depends: [
+  "conf-rocksdb"
+  "conf-pkg-config" {build}
+  (("host-arch-x86_32" {os = "win32" & os-distribution != "cygwinports"} & "conf-mingw-w64-zstd-i686" {os = "win32" & os-distribution != "cygwinports"}) |
+   ("host-arch-x86_64" {os = "win32" & os-distribution != "cygwinports"} & "conf-mingw-w64-zstd-x86_64" {os = "win32" & os-distribution != "cygwinports"}))
+]
+build: [
+
+  ["pkgconf" {os = "win32" & os-distribution = "cygwin"}
+   "--personality=i686-w64-mingw32" {os = "win32" & os-distribution = "cygwin" & host-arch-x86_32:installed}
+   "--personality=x86_64-w64-mingw32" {os = "win32" & os-distribution = "cygwin" & host-arch-x86_64:installed}
+   "pkg-config" {os != "win32" | os-distribution != "cygwin"}
+   "--atleast-version=7.8.3" "rocksdb"] {
+    !(os-distribution = "debian" & (os-version = 11 | os-version = 12)) &
+    !(os-distribution = "ubuntu" & (os-version = "20.04" | os-version = "22.04"))
+   }
+]
+synopsis: "Virtual package relying on a system installation of RocksDB (version >= 5.17.2)"
```

Comment:
I understand the need for this kind of package, and I wish we had a good solution (or at least a standard solution) to offer but right nnow it's an unsolved problem for the opam repo.

I'm not very happy with adding version _constraints_ as part of the name of a package. Although I can see how it's needed and how we can't reasonably generate conf-rocksdb packages for each available version of rocksdb. Or maybe that's reasonable but would require having a separate dedicate opam repository.

Anyway… Anyone has ideas?

Otherwise I guess we can possibly merge for now and possibly revisit.

Reply:
It's actually worse and worse.  When I started with trying to get this to work, I foolishly assumed that `pkg-config` would be available to check rocksdb's package-version.  But (haha, silly me) that wasn't the case with these older versions of rocksdb like (sigh) current Debian (11, 12) and backlevel Ubuntu (20.24, 22.04).  And they were all backlevel versions of Rocksdb (5.17.2, 6.11.4).   Other platforms had 7.8.3, though a few even failed the check.

I could put these changes into conf-rocksdb ?

The bigger problem of not being able to state version-constraints is .... painful.  Some packages (like Rocksdb) appear to be changing their API pretty rapidly.  But OTOH this is a big change to opam, and I sure think there are better ways to spend developer resource than on this.

Reply:
I think that we can accept the `ge-…` name for now

And maybe someone with time available can write a generator for many conf-packages where the version number is just substituted.